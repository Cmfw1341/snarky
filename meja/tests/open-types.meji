type t = ..;  type t += A  type t += B  let a : t;  let b : t; 
module T : { type t = ..;  type T.t += C  };  let c : T.t;  type T.t += D 
let d : T.t;  type u('a, 'b) = ..;  type u('a, 'b) += F('a) | E('a, 'b) 
let e : u(int, bool);  let f : u(int, int);  let g : u(unit, bool); 

