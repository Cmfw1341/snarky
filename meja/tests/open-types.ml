open Snarky
open Snarky.Snark
module Impl =
  Snarky.Snark.Run.Make (Snarky.Backends.Mnt4.Default) (Core_kernel.Unit)
open Impl

type t = ..

type t += A

type t += B

let (a : t) = A

let (b : t) = B

module T = struct
  type t = ..

  type t += C
end

let (c : T.t) = T.C

type T.t += D

let (d : T.t) = D

type ('a, 'b) u = ..

type ('a, 'b) u += E of 'a * 'b | F of 'a

let (e : (int, bool) u) = E (1, true)

let (f : (int, int) u) = E (1, 1)

let (g : (unit, bool) u) = F ()
