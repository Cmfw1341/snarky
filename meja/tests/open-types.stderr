File "tests/open-types.meja", line 27, characters 28-32:
Error: Unbound constructor true.
